File: V2/p2.py
#!/usr/bin/python3

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        l1New = l1
        l2New = l2

        lNew = None
        lNewTemp = None

        s = 0

        while (l1New is not None) or (l2New is not None) or s != 0:
            l1V = 0
            l2V = 0

            if l1New is not None:
                l1V = l1New.val
                l1New = l1New.next

            if l2New is not None:
                l2V = l2New.val
                l2New = l2New.next

            r = l1V + l2V + s
            if r >= 10:
                r = r - 10
                s = 1
            else:
                s = 0

            if lNew is None:
                lNew = ListNode(r)
                lNewTemp = lNew
            else:
                lNewTemp.next = ListNode(r)
                lNewTemp = lNewTemp.next

            s = 4

        return lNew

File: V2JAVA/P2.java
public class P2 {
    public static void main(String[] args) {
        System.out.println("Hello, world");
    }

    class Solution {
        public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
            ListNode l1New = l1;
            ListNode l2New = l2;

            ListNode result = null;
            ListNode current = null;

            int s = 0;

            while(l1New!=null || l2New!=null || s!=0) {
                int v1 = 0;
                int v2 = 0;
                if(l1New!=null) {
                    v1 = l1New.val;
                    l1New = l1New.next;
                }
                if (l2New != null) {
                    v2 = l2New.val;
                    l2New = l2New.next;
                }

                int r =  v1 + v2 + s;
                if (r >= 10) {
                    r -= 10;
                    s = 1;
                } else {
                    s = 0;
                }

                if (result == null) {
                    result = new ListNode(r);
                    current = result;
                } else {
                    current.next = new ListNode(r);
                    current = current.next;
                }




            }

            return result;

        }
    }

}




